packages/responsive-hooks/README.md

```diff
@@ -0,0 +1,39 @@
+# @theme-ui/responsive-hooks
+
+React hooks for referencing theme-ui breakpoints.
+
+## Overview
+
+This package provides a `useBreakpointIndex` React hook for referencing responsive values based on component's breakpoints.
+
+## Installation
+
+```sh
+npm i @theme-ui/responsive-hooks
+```
+
+## Creation
+
+```js
+import { useBreakpointIndex } from '@theme-ui/responsive-hooks'
+
+const ResponsiveText = props => {
+  const ref = useRef(null)
+  const index = useResponsiveIndex(props.breakpoints, ref.current)
```

Comment:
It'd be great if the hook here used `useThemeUI` internally to get the breakpoints from the theming context, but allow for bespoke overrides with an argument. I think the hook could also call `useRef` internally, so maybe the API could look something like: `const [ref, index] = useContainerQuery()` or `const [ref, index] = useContainerQuery([ '32em', '48em' ])`

Reply:
I've implemented this API, thanks for the suggestion, it's much cleaner. 

There are two things we have to clarify:

1. How to convert 'em' and 'rem' to px (ResizeObserver returns a width in pixels), it wasn't a case with CSS media-queries, but now we implement it with JS. AFAIK there are no decent solutions except some w3c drafts on how to convert em to px. I've provided my own code for it (requires to access DOM once).

2.  It works with all properties except **width** (a problem with a circular dependency):

If we provide a width based on a breakpoint index, then ResizeObserver calls a handler (because the width has been changed), then it will return a new index based on a new width, which leads to another ResizeObserver call, and again.

I can focus on tests and after that work on the circular dependency problem. What do you think @jxnblk

Reply:
I think _width_ cannot depend on the breakpoints. Example:
```
<ResponsiveText breakpoints={['480px']} widths={['100%', '25%']}>
  Hello
</ResponsiveText>
```
We use the next algorithm for getting a breakpoint index:
1. Take breakpoint index 0 as an initial value
2. Apply width as widths[0], it means '100%'
3. If the width more than 480px now, then apply a breakpoint width index 1
4. Apply width as widths[1], it means '25%'
5. If the width less than 480px now, then apply a breakpoint width index 0
6. Go to point 2. Cycle

All styles depend on the breakpoints prop and the breakpoints prop doesn't depend on them. Except for _width_ prop: breakpoints also depend on them.

Reply:
More explanation here:
https://twitter.com/philwalton/status/1220790253956395008

Maybe be can just implement custom breakpoints for the component based on the viewport width? API remains exactly the same.